packages/set/src/index.ts

```diff
@@ -18,74 +18,81 @@ const $KEYS = Symbol("track-keys");
 export class ReactiveSet<T> extends Set<T> {
   #triggers = new TriggerCache<T | typeof $KEYS>();
 
-  constructor(values?: Iterable<T> | null) {
+  constructor(values?: readonly T[] | null) {
     super();
-    if (values) for (const v of values) super.add(v);
+    if (values) for (const value of values) super.add(value);
+  }
+
+  [Symbol.iterator](): IterableIterator<T> {
+    return this.values();
   }
 
-  // reads
   get size(): number {
     this.#triggers.track($KEYS);
     return super.size;
   }
-  has(v: T): boolean {
-    this.#triggers.track(v);
-    return super.has(v);
+
+  has(value: T): boolean {
+    this.#triggers.track(value);
+    return super.has(value);
   }
 
-  *keys(): IterableIterator<T> {
-    for (const key of super.keys()) {
-      this.#triggers.track(key);
-      yield key;
-    }
-    this.#triggers.track($KEYS);
+  keys(): IterableIterator<T> {
+    return this.values();
   }
-  values(): IterableIterator<T> {
-    return this.keys();
+
+  *values(): IterableIterator<T> {
```

Comment:
Iterators no longer track specific keys. That optimization was working under the assumption that a BREAK will always happen in specific iteration, which might not be always a case as it can depend on external logic. All iterators should always be rerun if any op to the Set will happen.

Reply:
Can you mention that in the docs?

Reply:
Sure, Could you point me a place where I should do that?

Reply:
Not sure I understand what is expected here to document to be honest. There is no break optimization anymore, and the previous was a bug. Should't that be mentioned rather in changelog?